### PR TITLE
Feature/improve encryption perf

### PR DESCRIPTION
**Pull request by TobTheRock**

### CODE REVIEW

File: src/crypto/aead.rs [hunk 1]
```diff
@@ -46,43 +46,41 @@ mod ring {
...
+        fn generate_iv(
+            mut counter: impl Iterator<Item = u8>,
+            salt: &[u8],
+        ) -> [u8; ring::aead::NONCE_LEN] {
+            let mut iv = [0u8; ring::aead::NONCE_LEN];
+            for i in (0..ring::aead::NONCE_LEN).rev() {
+                iv[i] = salt[i];
+                if let Some(counter_byte) = counter.next() {
+                    iv[i] = iv[i] ^ counter_byte;
+                }
```
[the remark this answers is not in the document]
Comment (hoodie) [reply src/crypto/aead.rs:73]: you might want to add a commit that addresses these warnings

**Review by hoodie — DISMISSED** [review]

File: src/crypto/aead.rs [hunk 2]
```diff
@@ -189,7 +191,11 @@ mod ring {
                     let secret = KeyMaterial(&test_vector.key_material)
                         .expand_as_secret(&cipher_suite)
                         .unwrap();
-                    let nonce = FrameNonceSequence::new(test_vector.frame_count, &secret.salt);
+                    let nonce = FrameNonceSequence::new(
+                        &FrameCount::from(test_vector.frame_count),
```
Comment (hoodie) [inline src/crypto/aead.rs:195]: this is a bit weird with the temporaries. Do we maybe just want to do this with an `.into()` ?

File: src/header/extended_header.rs [hunk 3]
```diff
@@ -41,7 +41,7 @@ impl HeaderFields for ExtendedHeader {
 
     fn size(&self) -> usize {
         ExtendedHeader::STATIC_HEADER_LENGHT_BYTE
-            + get_nof_non_zero_bytes(self.key_id).max(1) as usize
+            + min_len_in_bytes(self.key_id) as usize
```
Comment (hoodie) [inline src/header/extended_header.rs:44]: can we make `min_len_in_bytes()` return a `usize` already?

**Review by hoodie — APPROVED** [review]

File: src/header/extended_header.rs (hunk 3 above)
Comment (TobTheRock) [reply src/header/extended_header.rs:44]: Unfortunately, we need it also as u8 when setting the bitfield? Maybe a new tuple type which can be interpreted as both u8 and usize could be the solution?

File: src/crypto/aead.rs (hunk 2 above)
Comment (TobTheRock) [reply src/crypto/aead.rs:195]: I will check that

File: src/header/extended_header.rs (hunk 3 above)
Comment (hoodie) [reply src/header/extended_header.rs:44]: naaa, its fine this way then

**Pull request merged**